packages/typography/src/Link/BackLink/BackLink.spec.tsx

```diff
@@ -0,0 +1,123 @@
+
+type SpanLikeProps = InferredPolymorphicPropsWithRef<'span', BaseBackLinkProps>;
+
+type AnchorLikeProps = InferredPolymorphicPropsWithRef<'a', BaseBackLinkProps>;
+
+type BackLinkRenderProps = AnchorLikeProps | SpanLikeProps;
+
+const renderBackLink = (props: BackLinkRenderProps) => {
+  const utils = render(
+    <BackLink data-testid="link" {...props}>
+      Back link
+    </BackLink>,
+  );
+  const link = utils.getByTestId('link');
+
+  return { ...utils, link };
+};
+
+describe('packages/typography', () => {
+  describe('BackLink', () => {
+    test('accepts a ref', () => {
+      const ref = createRef<HTMLAnchorElement>();
+      const { container } = renderBackLink({
+        ref,
+        href: 'http://mongodb.design',
+      });
+
+      expect(ref.current).toBe(container.firstElementChild);
+    });
+
+    test('renders name of link', () => {
+      const { getByText } = renderBackLink({
+        href: 'http://mongodb.design',
+      });
+      const text = getByText('Back link');
+      expect(text).toBeInTheDocument();
+    });
+
+    describe('inferred polymorphic behavior', () => {
+      test('when the "as" prop is supplied its value is respected', () => {
+        const { link } = renderBackLink({
+          href: 'http://localhost:9001',
+          /* @ts-expect-error to show that "as" overrides "href" */
+          as: 'div',
+        });
+
+        expect(link.tagName.toLowerCase()).toBe('div');
+      });
+
+      test('renders as an anchor when an href prop is supplied', () => {
+        const { link } = renderBackLink({ href: 'http://mongodb.design' });
+
+        expect(link.tagName.toLowerCase()).toBe('a');
+      });
+
+      test('renders as a span when no "as" or "href" is provided', () => {
+        const { link } = renderBackLink({});
+
+        expect(link.tagName.toLowerCase()).toBe('span');
+      });
+
+      test('renders with wrapper component', () => {
+        const Wrapper = (props: JSX.IntrinsicElements['a']) => {
+          return (
+            <a data-testid="link" {...props}>
+              wrapper component
+            </a>
+          );
+        };
+
+        const { getByTestId, getByText } = render(
+          <BackLink href="http://mongodb.design" as={Wrapper}>
+            Link
+          </BackLink>,
+        );
+        const link = getByTestId('link');
+        expect(link.getAttribute('href')).toBe('http://mongodb.design');
+        expect(link.tagName.toLowerCase()).toBe('a');
+
+        const text = getByText('wrapper component');
+        expect(text).toBeInTheDocument();
+      });
+    });
+
+    describe('TypeScript types are correct', () => {
+      const WrapperComponent = (props: JSX.IntrinsicElements['button']) => {
+        return <button {...props} />;
+      };
+
+      const AnchorComponent = (props: JSX.IntrinsicElements['a']) => {
+        return <a {...props}>test</a>;
+      };
+
+      // eslint-disable-next-line
+      test.skip('BackLink Component types', () => {
```

Comment:
can we test that it accepts target/rel attributes when anchor is set thorugh `as=AnchorComponent`, `as="a'`, and `href="string"`